pages/api/pearl-daa.js

```diff
@@ -0,0 +1,216 @@
+import {
+  CACHE_DURATION_SECONDS,
+  PREDICT_AGENT_CLASSIFICATION,
+  PREDICT_STAKING_PROGRAMS_PEARL,
+} from 'common-util/constants';
+import {
+  REGISTRY_GRAPH_CLIENTS,
+  STAKING_GRAPH_CLIENTS,
+} from 'common-util/graphql/client';
+import {
+  checkpointsQuery,
+  dailyBabydegenPerformancesQuery,
+  predictDailyAgentPerformancesQuery,
+} from 'common-util/graphql/queries';
+import { getMidnightUtcTimestampDaysAgo } from 'common-util/time';
+
+const getContractsToServiceId = (checkpoints) => {
+  return checkpoints.reduce((acc, checkpoint) => {
+    checkpoint.serviceIds.forEach((serviceId) => {
+      if (!acc[serviceId]) {
+        acc[serviceId] = [];
+      }
+      acc[serviceId].push(checkpoint.contractAddress);
+    });
+    return acc;
+  }, {});
+};
+
+const getPerformanceByAgentId = async () => {
+  // Only use valory_trader agent IDs
+  const traderAgentIds = PREDICT_AGENT_CLASSIFICATION.valory_trader.map((n) =>
+    Number(n),
+  );
```

Comment:
```suggestion
  const traderAgentIds = PREDICT_AGENT_CLASSIFICATION.valory_trader
```

those are already numbers